test/teos/unit/test_internal_api.py

```diff
@@ -232,6 +248,139 @@ def test_get_appointment_non_existent(internal_api, stub):
+
+
+def test_get_tower_info_empty(clear_state, internal_api, stub):
+    response = stub.get_tower_info(Empty())
+    assert isinstance(response, GetTowerInfoResponse)
+    assert response.tower_id == Cryptographer.get_compressed_pk(internal_api.watcher.signing_key.public_key)
+    assert response.n_registered_users == 0
+    assert response.n_watcher_appointments == 0
+    assert response.n_responder_trackers == 0
+
+
+def test_get_tower_info(internal_api, stub, monkeypatch):
+    monkeypatch.setattr(internal_api.watcher.gatekeeper, "registered_users", {"uid1": {}})
+    monkeypatch.setattr(
+        internal_api.watcher,
+        "appointments",
+        {
+            "uid1": {"locator": "locator1", "user_id": "user_id1"},
+            "uid2": {"locator": "locator2", "user_id": "user_id2"},
+        },
+    )
+    monkeypatch.setattr(
+        internal_api.watcher.responder,
+        "trackers",
+        {
+            "uid1": {"penalty_txid": "txid1", "locator": "locator1", "user_id": "user_id1"},
+            "uid2": {"penalty_txid": "txid2", "locator": "locator2", "user_id": "user_id2"},
+            "uid3": {"penalty_txid": "txid3", "locator": "locator2", "user_id": "user_id3"},
+        },
+    )
+
+    response = stub.get_tower_info(Empty())
+    assert isinstance(response, GetTowerInfoResponse)
+    assert response.tower_id == Cryptographer.get_compressed_pk(internal_api.watcher.signing_key.public_key)
+    assert response.n_registered_users == 1
+    assert response.n_watcher_appointments == 2
+    assert response.n_responder_trackers == 3
+
+
+def test_get_users(internal_api, stub, monkeypatch):
+    # it doesn't matter they are not valid user ids for the test
+    mock_users = ["user1", "user2", "user3"]
+    monkeypatch.setattr(
+        internal_api.watcher.gatekeeper, "registered_users", {"user1": dict(), "user2": dict(), "user3": dict()}
+    )
+
+    response = stub.get_users(Empty())
+    assert isinstance(response, GetUsersResponse)
+    assert response.user_ids == mock_users
+
+
+def test_get_user(internal_api, stub, monkeypatch):
+    # it doesn't matter they are not valid user ids and user data object for this test
+    mock_user_id = "some_user_id"
+    mock_user_info = UserInfo(100, 1234)
+
+    def mock_get_user_info(user_id):
+        if user_id == mock_user_id:
+            return mock_user_info
+        else:
+            raise f"called with an unexpected user_id: {user_id}"
+
+    monkeypatch.setattr(internal_api.watcher, "get_user_info", mock_get_user_info)
+
+    response = stub.get_user(GetUserRequest(user_id=mock_user_id))
+    assert isinstance(response, GetUserResponse)
+
+    assert json_format.MessageToDict(response.user) == mock_user_info.to_dict()
+
+
+def test_get_user_not_found(internal_api, stub):
+    mock_user_id = "some_non_existing_user_id"
+
+    with pytest.raises(grpc.RpcError) as e:
+        response = stub.get_user(GetUserRequest(user_id=mock_user_id))
```

Comment:
No need to store the call in a variable.

Reply:
Done in 7e8482d915bd416d957abeda3a0d0e8746bc5e94.